qutip/solver/multitraj.py

```diff
@@ -47,7 +73,11 @@ class MultiTrajSolver(Solver):
     }
 
     def __init__(self, rhs, *, options=None):
-        self.rhs = rhs
+        if isinstance(rhs, QobjEvo):
+            self.system = _MTSystem(rhs)
+        else:
```

Comment:
Could we check here that `rhs` really is an instance of `_MTSystem`? I guess someone could pass anything for the RHS accidentally and it would be nice to raise and error straight away.